feeds/scheduler/feed_group.go

```diff
@@ -125,14 +124,21 @@ func (fg *FeedGroup) publishPackages(pkgs []*feeds.Package) (int, error) {
 		}).Print("Sending package upstream")
 		b, err := json.Marshal(pkg)
 		if err != nil {
-			log.Printf("Error marshaling package: %#v", pkg)
-			return processed, err
+			log.WithField("name", pkg.Name).WithError(err).Error("Error marshaling package")
+			errs = append(errs, err)
 		}
 		if err := (fg.publisher).Send(context.Background(), b); err != nil {
-			log.Printf("Error sending package to upstream publisher %v", err)
-			return processed, err
```

Comment:
This could lead to a situation in which every publish call has to wait for the default timeout (unsure what it is) if say the publisher endpoint is down. I'm not sure this needs to be handled more gracefully in this PR, but it's definitely something to consider.

Reply:
The timeout appears to be publisher specific with no generalised options for configuring this without using the publisher specific constructors. In the case of kafka this uses sarama for the kafka client implementation, this uses 30s defaults for all read/write timeouts. However this may end up being a 30s timeout for EACH package published.

As for risk reduction, handling publishing in requests would lead to different behaviour where the HTTP response is before the data is published.